test/unit/spec-testing.ts

```diff
@@ -0,0 +1,55 @@
+// Copyright IBM Corp. 2017. All Rights Reserved.
+// Node module: loopback-next-hello-world
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+import {Application, CoreBindings} from '@loopback/core';
+var Dredd = require('dredd');
+import {expect} from '@loopback/testlab';
+import {HelloWorldApp} from '../..';
+
+describe('Api Spec End Points', () => {
+  let dredd: any;
+  before(initEnvironment);
+
+  describe('input/output test', () => {
+
+    it('passes match', done => {
+      dredd.run((err: Error, stats: object) => {
+        if (err) return done(err);
+        expect(stats).to.containDeep({
+          failures: 0,
+          errors: 0,
+          skipped: 0,
+        });
+        done();
+      });
+    });
+  });
+
+  async function initEnvironment() {
+    // By default, the port is set to 3000.
+    const app: Application = new HelloWorldApp();
+    // For testing, we'll let the OS pick an available port by setting
+    // CoreBindings.HTTP_PORT to 0.
+    app.bind(CoreBindings.HTTP_PORT).to(0);
+    // app.start() starts up the HTTP server and binds the acquired port
+    // number to CoreBindings.HTTP_PORT.
+    await app.start();
+    // Get the real port number.
+    const port: number = await app.get(CoreBindings.HTTP_PORT);
+    console.log(`Spec Test server listening on port ${port}`);
+    console.log(`Run \'curl localhost:${port}/helloworld?name=YOUR_NAME\' to try it out`);
```

Comment:
Please remove these two console logs, the test suite should finish fast, before the user has any chance to actually request this URL.